In [1]:
import numpy as np

Задание 1. Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. Посчитать (желательно без использования статистических методов наподобие std, var, mean) среднее арифметическое, среднее квадратичное отклонение, смещенную и несмещенную оценки дисперсий для данной выборки.

$$M(X) = \frac{1}{n} \sum\limits_{i=1}^{n} x_i$$

Среднее арифметическое:

где $n = 20$ количество выпускников в выборке.

$x=\frac{100+80+75+77+89+33+45+25+65+17+30+24+57+55+70+75+65+84+90+150}{20}=\frac{1306}{20}$

Среднее квадратичное отклонение:

$$\sigma = \sqrt{\frac{\sum\limits_{i=1}^{n} (x_i - \overline{x})^2}{n}}$$

$$\sigma = \sqrt{\frac{(100-65.3)^2+(80-65.3)^2+(75-65.3)^2+(77-65.3)^2+(89-65.3)^2+(33-65.3)^2+(45-65.3)^2+(25-65.3)^2+(65-65.3)^2+(17-65.3)^2+(30-65.3)^2+(24-65.3)^2+(57-65.3)^2+(55-65.3)^2+(70-65.3)^2+(75-65.3)^2+(65-65.3)^2+(84-65.3)^2+(90-65.3)^2+(150-65.3)^2}{20}}=\sqrt{\frac{(34.7)^2+(14.7)^2+(9.7)^2+(11.7)^2+(23.7)^2+(-32.3)^2+(-20.3)^2+(-40.3)^2+(-0.3)^2+(-48.3)^2+(-35.3)^2+(-41.3)^2+(-8.3)^2+(-10.3)^2+(4.7)^2+(9.7)^2+(-0.3)^2+(18.7)^2+(24.7)^2+(84.7)^2}{20}}=\sqrt{\frac{1204.09+216.09+94.09+136.89+561.69+1043.29+412.09+1624.09+0.09+2332.89+1246.09+1705.69+68.89+106.09+22.09+94.09+0.09+349.69+610.09+7174.09}{20}}=\sqrt{\frac{19002.2}{20}}=\sqrt{950.11}=30.82$$

In [2]:
a=1204.09+216.09+94.09+136.89+561.69+1043.29+412.09+1624.09+0.09+2332.89+1246.09+1705.69+68.89+106.09+22.09+94.09+0.09+349.69+610.09+7174.09
a

19002.2

In [3]:
np.sqrt(19002.2/20)

30.823854398825596

$$\sigma^2 = \frac{\sum\limits_{i=1}^{n} (x_i - \overline{x})^2}{n}$$
$$\sigma^2_{несмещ.} = \frac{\sum\limits_{i=1}^{n} (x_i - \overline{x})^2}{n - 1}$$

$$\sigma^2_{несмещ.}=\sqrt{\frac{19002.2}{19}}=\sqrt{\frac{5 \cdot 19002+1}{5} \cdot \frac{1}{19}}=\sqrt{\frac{95010+1}{5} \cdot \frac{1}{19}}=\sqrt{950.11}=\sqrt{\frac{95011}{5 \cdot 19}}=\sqrt{\frac{95011}{95}}=\sqrt{1000.11579}=31.62$$

In [4]:
np.sqrt(95011/95)

31.624607341019814

In [5]:
a = np.array([100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150])
print(f'Массив = {a}')
print(f'Длина массива = {a.size}')
print(f'Функция mean(a) = {round(np.mean(a),2)} вычисляет среднее арифметическое значений элементов массива.')
print(f'Функция var(a) = {round(np.var(a),2)} вычисляет дисперсию значений элементов массива.')
print(f'Функция std(a) = {round(np.std(a),2)} вычисляет среднеквадратичное (стандартное) отклонение значений элементов массива.')

Массив = [100  80  75  77  89  33  45  25  65  17  30  24  57  55  70  75  65  84
  90 150]
Длина массива = 20
Функция mean(a) = 65.3 вычисляет среднее арифметическое значений элементов массива.
Функция var(a) = 950.11 вычисляет дисперсию значений элементов массива.
Функция std(a) = 30.82 вычисляет среднеквадратичное (стандартное) отклонение значений элементов массива.


Задание 2. В первом ящике находится 8 мячей, из которых 5 - белые. Во втором ящике - 12 мячей, из которых 5 белых. Из первого ящика вытаскивают случайным образом два мяча, из второго - 4. Какова вероятность того, что 3 мяча белые?

Вариантов комбинаций несколько:

(1) бб (2) бччч $ p_{1^1}= \frac{5}{8} \cdot \frac{4}{7},  p_{2^1}= \frac{5}{12} \cdot \frac{7}{11} \cdot \frac{6}{10} \cdot \frac{5}{9}, P_{1}=p_{1^1}*p_{2^1}=\frac{5}{8} \cdot \frac{4}{7} * \frac{5}{12} \cdot \frac{7}{11} \cdot \frac{6}{10} \cdot \frac{5}{9} = \frac{5 \cdot 4 \cdot 5 \cdot 7 \cdot 6 \cdot 5}{8 \cdot 7 \cdot 12 \cdot 11 \cdot 10 \cdot 9}$

(1) бч (2)ббчч $ p_{1^2}= \frac{5}{8} \cdot \frac{3}{7},  p_{2^2}= \frac{5}{12} \cdot \frac{4}{11} \cdot \frac{7}{10} \cdot \frac{6}{9}, P_{2}=p_{1^2}*p_{2^2}=\frac{5}{8} \cdot \frac{3}{7} * \frac{5}{12} \cdot \frac{4}{11} \cdot \frac{7}{10} \cdot \frac{6}{9} = \frac{5 \cdot 3 \cdot 5 \cdot 4 \cdot 7 \cdot 6}{8 \cdot 7 \cdot 12 \cdot 11 \cdot 10 \cdot 9}$

(1) чч (2)бббч $ p_{1^3}= \frac{3}{8} \cdot \frac{2}{7},  p_{2^3}= \frac{5}{12} \cdot \frac{4}{11} \cdot \frac{3}{10} \cdot \frac{7}{9}, P_{3}=p_{1^3}*p_{2^3}=\frac{3}{8} \cdot \frac{2}{7} * \frac{5}{12} \cdot \frac{4}{11} \cdot \frac{3}{10} \cdot \frac{7}{9} = \frac{3 \cdot 2 \cdot 5 \cdot 4 \cdot 3 \cdot 7}{8 \cdot 7 \cdot 12 \cdot 11 \cdot 10 \cdot 9}$

Общие варианты комбинаций:

(1)бб (2)бччч
<br>(1)бб (2)чбчч
<br>(1)бб (2)ччбч
<br>(1)бб (2)чччб - 4 комбинации, вероятность каждой комбинации одинакова, можем умножить вероятность первой на 4

(1)бч (2)ббчч
<br>(1)бч (2)бчбч
<br>(1)бч (2)бччб
<br>(1)бч (2)ччбб
<br>(1)бч (2)чбчб
<br>(1)бч (2)чббч 
<br>(1)чб (2)ббчч
<br>(1)чб (2)бчбч
<br>(1)чб (2)бччб
<br>(1)чб (2)ччбб
<br>(1)чб (2)чбчб
<br>(1)чб (2)чббч - 12 комбинаций, вероятность каждой комбинации одинакова, можем умножить вероятность первой на 12

(1)чч (2)бббч
<br>(1)чч (2)бчбб
<br>(1)чч (2)ббчб
<br>(1)чч (2)чббб - 4 комбинации, вероятность каждой комбинации одинакова, можем умножить вероятность первой на 4.

вероятность для первых посчитали выше.



$P=4 \cdot \frac{5 \cdot 4 \cdot 5 \cdot 7 \cdot 6 \cdot 5}{8 \cdot 7 \cdot 12 \cdot 11 \cdot 10 \cdot 9}+12 \cdot \frac{5 \cdot 3 \cdot 5 \cdot 4 \cdot 7 \cdot 6}{8 \cdot 7 \cdot 12 \cdot 11 \cdot 10 \cdot 9}+4 \cdot \frac{3 \cdot 2 \cdot 5 \cdot 4 \cdot 3 \cdot 7}{8 \cdot 7 \cdot 12 \cdot 11 \cdot 10 \cdot 9}=\frac{4 \cdot 5 \cdot 4 \cdot 5 \cdot 7 \cdot 6 \cdot 5 + 12 \cdot 5 \cdot 3 \cdot 5 \cdot 4 \cdot 7 \cdot 6 + 4 \cdot 3 \cdot 2 \cdot 5 \cdot 4 \cdot 3 \cdot 7}{8 \cdot 7 \cdot 12 \cdot 11 \cdot 10 \cdot 9}=\frac{1 \cdot 5 \cdot 4 \cdot 5 \cdot 7 \cdot 1 \cdot 1 + 1 \cdot 1 \cdot 3 \cdot 5 \cdot 2 \cdot 7 \cdot 6 + 1 \cdot 1 \cdot 1 \cdot 4 \cdot 3 \cdot 7}{8 \cdot 7 \cdot 1 \cdot 11 \cdot 1 \cdot 9}=\frac{5 \cdot 4 \cdot 5 \cdot 7  + 3 \cdot 5 \cdot 2 \cdot 7 \cdot 6 + 4 \cdot 3 \cdot 7}{8 \cdot 7 \cdot 11 \cdot 9}= \frac{700+1260+84}{8 \cdot 7 \cdot 11 \cdot 9}=\frac{2044}{8 \cdot 7 \cdot 11 \cdot 9}=\frac{511}{2 \cdot 7 \cdot 11 \cdot 9}=\frac{73}{2 \cdot 11 \cdot 9}=\frac{73}{198}=0,36868$

In [6]:
print(f'{round(73/198*100, 2)}%')

36.87%


Задание 3. В университет на факультеты A и B поступило равное количество студентов, а на факультет C студентов поступило столько же, сколько на A и B вместе. Вероятность того, что студент факультета A сдаст первую сессию, равна 0.8. Для студента факультета B эта вероятность равна 0.7, а для студента факультета C - 0.9. Студент сдал первую сессию. Какова вероятность, что он учится: 
<br>a). на факультете A 
<br>б). на факультете B 
<br>в). на факультете C?

$$P(A\:|\:B_1) = 0.8$$

$$P(A\:|\:B_2) = 0.7$$

$$P(A\:|\:B_3) = 0.9$$

$$P(B\:|\:A) = \frac{P(B) \cdot P(A\:|\:B)}{P(A)}$$

$$P(A)=P(B1)⋅P(A|B1)+P(B2)⋅P(A|B2)+...+P(Bn)⋅P(A|Bn)$$

a). на факультете A

$$P(B_{1}\:|\:A) = \frac{P(B_{1}) \cdot P(A\:|\:B_{1})}{P(A)}$$

$$P(A)=P(B_1)⋅P(A|B_1)+P(B_2)⋅P(A|B_2)+P(B_3)⋅P(A|B_3)$$

$$P(A)=\frac{1}{4}⋅0.8+\frac{1}{4}⋅0.7+\frac{1}{2}⋅0.9=\frac{0.8+0.7+2⋅0.9}{4}=\frac{3.3}{4}=\frac{33}{40}$$

$$P(B_{1}\:|\:A) = \frac{P(B_{1}) \cdot P(A\:|\:B_{1})}{P(A)}=\frac{P(B_{1}) \cdot P(A\:|\:B_{1})}{P(A)}=\frac{\frac{1}{4} \cdot \frac{4}{5}}{\frac{33}{40}}=\frac{\frac{1}{5}}{\frac{33}{40}}=\frac{1}{5} \cdot \frac{40}{33}=\frac{8}{33}=0.24$$

In [7]:
0.8+0.7+2*0.9

3.3

In [8]:
8/33

0.24242424242424243

б). на факультете В

$$P(A)=\frac{33}{40}$$

$$P(B_{2}\:|\:A) = \frac{P(B_{2}) \cdot P(A\:|\:B_{2})}{P(A)}=\frac{\frac{1}{4} \cdot \frac{7}{10}}{\frac{33}{40}}=\frac{\frac{7}{40}}{\frac{33}{40}}=\frac{7}{40} \cdot \frac{40}{33}=\frac{7}{33}=0.21$$

In [9]:
7/33

0.21212121212121213

в). на факультете C

$$P(A)=\frac{33}{40}$$

$$P(B_{3}\:|\:A) = \frac{P(B_{3}) \cdot P(A\:|\:B_{3})}{P(A)}=\frac{\frac{1}{2} \cdot \frac{9}{10}}{\frac{33}{40}}=\frac{\frac{9}{20}}{\frac{33}{40}}=\frac{9}{20} \cdot \frac{40}{33}=\frac{18}{33}=0.54$$

In [10]:
18/33

0.5454545454545454

Задача 4. Устройство состоит из трех деталей. Для первой детали вероятность выйти из строя в первый месяц равна 0.1, для второй - 0.2, для третьей - 0.25. Какова вероятность того, что в первый месяц выйдут из строя: 
<br>а). все детали <br>б). только две детали <br>в). хотя бы одна деталь <br>г). от одной до двух деталей?

$P_1=0.1$
<br>$P_2=0.2$
<br>$P_3=0.25$

a). break(1) break(2) break(3)

$P=P_1(A) \cdot P_2(A) \cdot P_3(A) =0.1 \cdot 0.2 \cdot 0.25 \cdot = \frac{1}{10} \cdot \frac{1}{5} \cdot \frac{1}{4}=\frac{1}{200}$

б). только две детали<br>break(1) break(2) not(3)
<br>$ p_{1}= \frac{1}{10} \cdot \frac{1}{5} \cdot (1-\frac{1}{4})$
<br>break(1) not(2) break(3)
<br>$ p_{2}= \frac{1}{10} \cdot (1-\frac{1}{5}) \cdot \frac{1}{4}$
<br>not(1) break(2) break(3)
<br>$ p_{2}= (1-\frac{1}{10}) \cdot \frac{1}{5} \cdot \frac{1}{4}$
<br>$P=p_1 + p_2 + p_3=\frac{1}{10} \cdot \frac{1}{5} \cdot (1-\frac{1}{4})+\frac{1}{10} \cdot (1-\frac{1}{5}) \cdot \frac{1}{4}+(1-\frac{1}{10}) \cdot \frac{1}{5} \cdot \frac{1}{4}=\frac{1}{10} \cdot \frac{1}{5} \cdot \frac{3}{4}+\frac{1}{10} \cdot \frac{4}{5} \cdot \frac{1}{4}+\frac{9}{10} \cdot \frac{1}{5} \cdot \frac{1}{4}=\frac{3+4+9}{10 \cdot 5 \cdot 4}=\frac{16}{10 \cdot 5 \cdot 4}=\frac{2}{5 \cdot 5 \cdot 1}=\frac{2}{25}=0.08$

In [11]:
2/25

0.08

в). хотя бы одна деталь

$ p_{allwork2gether}= (1-\frac{1}{10}) \cdot (1-\frac{1}{5}) \cdot (1-\frac{1}{4})=\frac{9}{10} \cdot \frac{4}{5} \cdot \frac{3}{4}=\frac{27}{50}$ тут мы посчитали вероятность что ни одна деталь не сломается

<br>Значит если вычтем это из 1 получим вероятность, при которой что-то не будет работать

$p_{oneatleast}=1 - \frac{27}{50}= \frac{50-27}{50}=\frac{23}{50}=0.46$

In [12]:
23/50

0.46

г). от одной до двух деталей?

Варианты для одной: <br>break(1) not(2) not(3)
<br>$ p_{1}= \frac{1}{10} \cdot (1-\frac{1}{5}) \cdot (1-\frac{1}{4})=\frac{1}{10} \cdot \frac{4}{5} \cdot \frac{3}{4}$
<br>not(1) break(2) not(3)
<br>$ p_{2}= (1-\frac{1}{10}) \cdot \frac{1}{5} \cdot (1-\frac{1}{4})=\frac{9}{10} \cdot \frac{1}{5} \cdot \frac{3}{4}$
<br>not(1) not(2) break(3)
<br>$ p_{2}= (1-\frac{1}{10}) \cdot (1-\frac{1}{5}) \cdot \frac{1}{4}=\frac{9}{10} \cdot \frac{4}{5} \cdot \frac{1}{4}$
<br>$P_1=p_1 + p_2 + p_3=\frac{1}{10} \cdot \frac{4}{5} \cdot \frac{3}{4}+\frac{9}{10} \cdot \frac{1}{5} \cdot \frac{3}{4}+\frac{9}{10} \cdot \frac{4}{5} \cdot \frac{1}{4}=\frac{12+27+36}{10 \cdot 5 \cdot 4}=\frac{75}{10 \cdot 5 \cdot 4}=\frac{3}{8}=0.375$

Варианты для двух считали выше в варианте б):
<br>$ P_2=0.08$

Тогда от одной до двух деталей: $P=P_1+P_2=0.375+0.08=0.455$

In [13]:
3/8

0.375

In [14]:
0.375+0.08

0.455